core/src/main/scala/caliban/validation/DefaultValue.scala

```diff
@@ -0,0 +1,153 @@
+package caliban.validation
+
+import caliban.CalibanError.ValidationError
+import caliban.InputValue.VariableValue
+import caliban.Value.NullValue
+import caliban.execution.{ ExecutionRequest, Field => F }
+import caliban.introspection.Introspector
+import caliban.introspection.adt._
+import caliban.introspection.adt.__TypeKind._
+import caliban.parsing.SourceMapper
+import caliban.parsing.adt.Definition.ExecutableDefinition.{ FragmentDefinition, OperationDefinition }
+import caliban.parsing.adt.Definition.{ TypeSystemDefinition, TypeSystemExtension }
+import caliban.parsing.adt.OperationType._
+import caliban.parsing.adt.Selection.{ Field, FragmentSpread, InlineFragment }
+import caliban.parsing.adt.Type.NamedType
+import caliban.parsing.adt._
+import caliban.schema.{ RootSchema, RootSchemaBuilder, RootType, Types }
+import caliban.{ InputValue, Rendering, Value }
+import caliban.parsing.Parser
+import zio.IO
+import caliban.schema.ArgBuilder
+import caliban.InputValue.ListValue
+import caliban.Value.EnumValue
+import caliban.Value.StringValue
+import caliban.Value.FloatValue.BigDecimalNumber
+import caliban.Value.IntValue.BigIntNumber
+import caliban.Value.BooleanValue
+import caliban.Value.IntValue.IntNumber
+import caliban.Value.IntValue.LongNumber
+import caliban.Value.FloatValue.DoubleNumber
+import caliban.Value.FloatValue.FloatNumber
+import caliban.InputValue.ObjectValue
+
+object DefaultValue {
```

Comment:
`DefaultValueValidator`? To keep the naming consistent

Reply:
Naming is my weakness :D

945e1e4